Sources/Purchasing/StoreKit2/ProductsFetcherSK2.swift

```diff
@@ -43,20 +42,25 @@ actor ProductsFetcherSK2 {
         }
     }
 
+    func clearCache() {
+        self.cachedProductsByIdentifier.removeAll(keepingCapacity: false)
+    }
+
 }
 
 @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
-private extension ProductsFetcherSK2 {
+extension ProductsFetcherSK2 {
+    /// Read-only variable declared for testing only.
+    var cachedProducts: [SK2StoreProduct] {
+        Array(self.cachedProductsByIdentifier.values)
```

Comment:
Why not make `cachedProductsByIdentifier)` as `private(set)`?

Reply:
Well, this was [my first approach](https://github.com/RevenueCat/purchases-ios/commit/f3c4c15ae3a6a13f039bda9f1577d441bb19805e).  However, I thought it might be better to have a read-only property for testing purposes, but yeah, I don't have a strong preference for that, I could make the getter function internal

WDYT?

Reply:
Totally agree. KIS principle! 😬

Reply:
Done ✅ ([d302f8f](https://github.com/RevenueCat/purchases-ios/pull/1583/commits/d302f8ffd66cc3d5feeed9bc694e5a90b80708bb))